client/new-stack-editor/lib/views/wizard/stackwizardmodal.coffee

```diff
@@ -0,0 +1,44 @@
+kd = require 'kd'
+EnvironmentFlux = require 'app/flux/environment'
+
+StackWizard = require './stackwizard'
+
+
+module.exports = class StackWizardModal extends kd.ModalView
+
+  constructor: (options = {}, data) ->
+
+    options          =
+      handleRoute    : options.handleRoute ? no
+      cssClass       : 'stack-wizard-modal'
+      width          : 820
+      overlay        : yes
+      overlayOptions :
+        cssClass     : 'second-overlay'
+
+    super options, data
+
+    wizard = new StackWizard
+
+    wizard.once 'ProviderSelected', (selectedProvider) =>
+      @createStackTemplate selectedProvider
+      @destroy()
+
+    wizard.on 'StackWizardCancelled', @bound 'destroy'
+
+    if @getOption 'handleRoute'
+      @on 'KDObjectWillBeDestroyed', -> kd.singletons.router.back()
+
+    @addSubView wizard
+
+
+  createStackTemplate: (selectedProvider) ->
+
+    { router } = kd.singletons
+
+    unless selectedProvider
```

Comment:
my bad, in the previous example there was a case for this one but it's not valid anymore.